### PR TITLE
RFC: RPCEndpoint standardization

### PR DESCRIPTION
**Pull request by volokluev**
### Context

This project will need observability, monitoring and other production readiness facets. This PR creates a wrapper around our RPC functionality to allow:

* default behavior for before and after each request execution (to allow for timing, reporting, rate limiting, etc)
* automatic registering of endpoints by name without having to modify global dictionaries
* listing all available endpoints (to allow for debugging tools such as in snuba admin)

### Desired feedback

* is the intent clear?
* Do the interfaces look ergonomic?
* Are the names appropriate?

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -30,9 +30,9 @@ def pytest_configure() -> None:
         settings.TESTING
     ), "settings.TESTING is False, try `SNUBA_SETTINGS=test` or `make test`"
 
-    initialize_snuba()
+    # initialize_snuba()
```
Comment (colin-sentry) [inline tests/conftest.py:33]: ??

File: tests/web/rpc/test_base.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from snuba.web.rpc import RPCEndpoint
+
+
+class MyRPC(RPCEndpoint[Timestamp, Timestamp]):
```
Comment (colin-sentry) [inline tests/web/rpc/test_base.py:6]: add a comment or something? unclear what this is for

Comment (colin-sentry) [reply tests/web/rpc/test_base.py:6]: this is to test that the registration works?

File: tests/web/rpc/v1alpha/test_trace_item_attribute_list.py [hunk 3]
```diff
@@ -5,17 +5,17 @@
 import pytest
 from google.protobuf.timestamp_pb2 import Timestamp
 from sentry_protos.snuba.v1alpha.endpoint_tags_list_pb2 import (
-    TraceItemAttributesRequest,
+    TraceItemAttributesRequest as TraceItemAttributesRequestProto,
```
Comment (colin-sentry) [inline tests/web/rpc/v1alpha/test_trace_item_attribute_list.py:8]: is annoying to have to do default imports like this, can we just make sentry-protos name things blahproto by default?

File: snuba/web/views.py [hunk 4]
```diff
@@ -273,14 +273,11 @@ def unqualified_query_view(*, timer: Timer) -> Union[Response, str, WerkzeugResp
 
 
 @application.route("/rpc/<name>/<version>", methods=["POST"])
-@util.time_request("rpc")
```
Comment (evanh) [inline snuba/web/views.py:276]: Why did you remove this?

**Review by evanh — APPROVED** [review]
LGTM
